src/lib/secrets/wallets.ml

```diff
@@ -0,0 +1,96 @@
+open Core
+open Async
+module Secret_keypair = Keypair
+open Signature_lib
+
+(* A simple cache on top of the fs *)
+type t = {mutable cache: Keypair.t list; path: string}
+
+(* TODO: Don't just generate bad passwords *)
+let password = lazy (Deferred.Or_error.return (Bytes.of_string ""))
+
+let load ~logger ~disk_location : t Deferred.t =
+  let logger =
+    Logger.extend logger [("wallets_context", `String "Wallets.get")]
+  in
+  let path = disk_location ^/ "store" in
+  let%bind () = File_system.create_dir path in
+  let%bind handle = Unix.opendir path in
+  let rec go () =
+    match%bind Unix.readdir_opt handle with
+    | None ->
+        return []
+    | Some next_file -> (
+        if
+          next_file = "." || next_file = ".."
+          || Filename.check_suffix next_file ".pub"
+        then go ()
+        else
+          match%bind
+            Secret_keypair.read ~privkey_path:(path ^/ next_file) ~password
+          with
+          | Ok keypair ->
+              let%bind xs = go () in
+              return (keypair :: xs)
+          | Error e ->
+              Logger.error logger ~module_:__MODULE__ ~location:__LOC__
+                "Failed to read %s: %s" path (Error.to_string_hum e) ;
+              go () )
+  in
+  let%bind () = Unix.chmod path ~perm:0o700 in
+  let%map keypairs = go () in
+  {cache= keypairs; path}
+
+(** Effectfully generates a new private key file and a keypair *)
+let generate_new t : Keypair.t Deferred.t =
+  let keypair = Keypair.create () in
+  let privkey_path =
+    Filename.concat t.path
```

Comment:
Can you use `Core.(^/)`?

Reply:
This one I intentionally didn't because the right-hand-side was fairly complex. Does that make sense, or do you think I should still use `^/`?